### PR TITLE
Offline Update (.jar replace)

### PR DESCRIPTION
**Pull request by gerth2**
Adds functionality to replace .jar files via settings page.

### CODE REVIEW

**Comment by gerth2** [comment]
Key Limitation: .jar files larger than ~60Mb will cause JVM out of memory errors on latest pi image.

Launching with `-Xmx512M` in the JVM arguments is the workaround used for testing. 

Per discord - likely we just want to change to use target-specific .jars to get under the size limit.

**Comment by gerth2** [comment]
Should there be additional checks on the uploaded .jar? Like, it's name contains `photonvision`, or it is a certain size, or has some specific contents to prove it's a photonvision .jar?

**Comment by jasondaming** [comment]
Closes #249

**Comment by gerth2** [comment]
NOTE: We probably shouldn't merge this until either A) pi image is updated to allow for more RAM usage, or B) the jar file is split out and reduced in size to ~60Mb

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -160,7 +160,7 @@ jobs:
       - run: |
           NEW_JAR=$(realpath $(find . -name photonvision\*.jar))
           sudo apt install unzip zip
-          curl -sk https://api.github.com/repos/photonvision/photon-pi-gen/releases/latest | grep "browser_download_url.*zip" | cut -d : -f 2,3 | tr -d '"' | wget -qi -
+          curl -sk https://api.github.com/repos/photonvision/photon-pi-gen/releases/tags/v2021.1.4 | grep "browser_download_url.*zip" | cut -d : -f 2,3 | tr -d '"' | wget -qi -
```
Comment (bankst-oxos) [inline .github/workflows/main.yml:163]: Was this change meant to be committed?

File: photon-server/src/main/resources/web/index.html [hunk 2]
```diff
@@ -1 +1 @@
-<p>UI has not been copied!</p>
\ No newline at end of file
+<!DOCTYPE html><html lang=en><head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width,initial-scale=1"><link rel=icon href=/favicon.png><title>PhotonVision</title><link href=/js/chunk-2d216214.4302abb0.js rel=prefetch><link href=/js/chunk-2d216257.0de4b8cc.js rel=prefetch><link href=/js/chunk-ad0384ec.7dafafc8.js rel=prefetch><link href=/css/app.8acbeb98.css rel=preload as=style><link href=/css/chunk-vendors.25378257.css rel=preload as=style><link href=/js/app.babd63e8.js rel=preload as=script><link href=/js/chunk-vendors.7a922cb5.js rel=preload as=script><link href=/css/chunk-vendors.25378257.css rel=stylesheet><link href=/css/app.8acbeb98.css rel=stylesheet></head><body><noscript><strong>We're sorry but PhotonVision doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id=app></div><script src=/js/chunk-vendors.7a922cb5.js></script><script src=/js/app.babd63e8.js></script></body></html>
```
Comment (bankst-oxos) [inline photon-server/src/main/resources/web/index.html]: Undo

**Review by bankst-oxos — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (Bankst) [inline .github/workflows/main.yml:163]: Was this meant to be committed?

File: photon-server/src/main/resources/web/index.html (hunk 2 above)
Comment (Bankst) [inline photon-server/src/main/resources/web/index.html:1]: Revert
```suggestion
<p>UI has not been copied!</p> 
```

**Review by Bankst — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (gerth2) [reply .github/workflows/main.yml:163]: Yes - per https://discord.com/channels/725836368059826228/725846807019782174/915598652934881322  and the conversation before it - goal is to allow photon to choose explicitly when it move up and down in pigen versions.

Also happens to resolve the weirdness of github calling the first tag on a particular node "Latest"

**Review by mcm001 — APPROVED** [review]

**Review by mcm001 — APPROVED** [review]

**Pull request merged**